### ISSUE TITLE
MVCArray bindings are broken

### ISSUE THREAD

**Issue by dietmarebner** [a14n/dart-google-maps#65]
Bindings for MVCArray expose three streams, i.e., onInsertAt, onRemoveAT, and onSetAt.

The first one is correct but the other two wrap the click event rather than 'remove_at' and 'set_at' respectively. This is in dart-google-maps/lib/src/google_maps_src.g.dart:2889.

There's a test for this in dart-google-maps/test/gmaps_test.dart but that doesn't actually fail if the event handlers don't get executed.

**Comment by a14n** [comment]
Thanks for filing an issue. _google_maps-3.3.3_ fixes it.

**Comment by dietmarebner** [comment]
Thanks, Alexandre. 

I think there's something missing with the codecs business. remove_at and set_at take two arguments (index and previous element) and return a IndexAndElement<E>. _codec32 only does ([dynamic]) → dynamic but not the two-argument version.

Something like the snippet below works but this looks like auto-generated code so this should probably fixed elsewhere.

class _Undefined {
  const _Undefined();
}

const _undefined = _Undefined();

final __codecOneOrTwoArgFunction =
    FunctionCodec<Function> /*<([dynamic, dynamic]) → dynamic>*/(
  (f) => ([p$p1, p$p2 = _undefined]) {
        var result = p$p2 == _undefined
            ? f(__codec9.decode(p$p1))
            : f(__codec9.decode(p$p1), __codec9.decode(p$p2));
        return __codec9.encode(result);
      },
  (f) => ([p$p1, p$p2 = _undefined]) {
        var args = [__codec9.encode(p$p1)];
        if (p$p2 != _undefined) args.add(__codec9.encode(p$p2));
        var result = f is JsFunction ? f.apply(args) : Function.apply(f, args);
        return __codec9.decode(result);
      },
);

**Comment by drewwarren** [comment]
As dietmarebner commented above subscribing to onRemoveAT and onSetAt are still broken. You can be replicate this by attempting to run the tests for MVCArray:

`pub run test -p chrome test/gmaps_test.dart`
...
```
00:06 +2: MVCArray works                                                                                           
inserted at 0
00:06 +2 -1: MVCArray works [E]                                                                                    
  NoSuchMethodError: method not found: 'call'
  Receiver: Closure '_closure0'
  Arguments: [0, "aa"]
  google3:///third_party/dart_lang/v2_3_3_dev_0_0/sdk/lib/_internal/js_runtime/lib/js_helper.dart 1274:37   Object.wrapException
  google3:///third_party/dart_lang/v2_3_3_dev_0_0/sdk/lib/_internal/js_runtime/lib/core_patch.dart 63:5     _closure0.noSuchMethod
  google3:///third_party/dart_lang/v2_3_3_dev_0_0/sdk/lib/_internal/js_runtime/lib/js_helper.dart 938:5     Object.Primitives.functionNoSuchMethod
  google3:///third_party/dart_lang/v2_3_3_dev_0_0/sdk/lib/_internal/js_runtime/lib/js_helper.dart 1108:9    Object.Primitives._genericApplyFunction2
  google3:///third_party/dart_lang/v2_3_3_dev_0_0/sdk/lib/_internal/js_runtime/lib/js_helper.dart 1037:12   Object.Primitives.applyFunction
  google3:///third_party/dart_lang/v2_3_3_dev_0_0/sdk/lib/_internal/js_runtime/lib/core_patch.dart 83:5     Object.Function.apply
  google3:///third_party/dart_lang/v2_3_3_dev_0_0/sdk/lib/js/dart2js/js_dart2js.dart 125:32                 _callDartFunction
  google3:///third_party/dart_lang/v2_3_3_dev_0_0/sdk/lib/js/dart2js/js_dart2js.dart 581:45                 _.fe.<fn>
```

**Comment by a14n** [comment]
@drewwarren Thanks for your feedback. The issue should be addressed by 0cbd411 publish in version 3.4.1
